smurf_parser.py

```diff
@@ -0,0 +1,80 @@
+from rply import ParserGenerator
+from ast import Number, Print, Sum, Sub, Mult, Div, Assign, Variable
+
+
+class Parser:
+    def __init__(self):
+        self.pg = ParserGenerator(
+            # A list of all token names accepted by the parser.
+            ['LET', 'NUMBER', 'PRINT', 'OPEN_PAREN', 'CLOSE_PAREN', 'MULT', 'DIV', 'SUM', 'SUB', 'ASSIGN', 'VAR', 'COMMA'],
+            precedence=[
+                ('left', ['COMMA']),
+                ('left', ['PRINT']),
+                ('left', ['ASSIGN']),
+                ('left', ['LET']),
+                ('left', ['SUM', 'SUB', ]),
+                ('left', ['MULT', 'DIV', ]),
+                ('left', ['OPEN_PAREN', 'CLOSE_PAREN', ]),
+            ]
+        )
+
+        self.variables = {}
+
+    def parse(self):
```

Comment:
I don't know rply, but it seems to me that you never tell it that a program can be a list of expressions: doesn't this just parse one and then stop?